pkg/kubectl/cmd/patch_test.go

```diff
@@ -92,3 +93,50 @@ func TestPatchObjectFromFile(t *testing.T) {
 		t.Errorf("unexpected output: %s", buf.String())
 	}
 }
+
+func TestPatchObjectFromFileOutput(t *testing.T) {
+	_, svc, _ := testData()
+
+	svcCopyObj, err := api.Scheme.DeepCopy(&svc.Items[0])
+	if err != nil {
+		t.Fatal(err)
+	}
+	svcCopy := svcCopyObj.(*api.Service)
+	if svcCopy.Labels == nil {
+		svcCopy.Labels = map[string]string{}
+	}
+	svcCopy.Labels["post-patch"] = "post-patch-value"
+
+	f, tf, codec, _ := cmdtesting.NewAPIFactory()
+	p := &testPrinter{}
+	tf.Printer = p
+	tf.UnstructuredClient = &fake.RESTClient{
+		APIRegistry:          api.Registry,
+		NegotiatedSerializer: unstructuredSerializer,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			switch p, m := req.URL.Path, req.Method; {
+			case p == "/namespaces/test/services/frontend" && m == "GET":
+				return &http.Response{StatusCode: 200, Header: defaultHeader(), Body: objBody(codec, &svc.Items[0])}, nil
+			case p == "/namespaces/test/services/frontend" && m == "PATCH":
+				return &http.Response{StatusCode: 200, Header: defaultHeader(), Body: objBody(codec, svcCopy)}, nil
+			default:
+				t.Fatalf("unexpected request: %#v\n%#v", req.URL, req)
+				return nil, nil
+			}
+		}),
+	}
+	tf.Namespace = "test"
+	buf := bytes.NewBuffer([]byte{})
+
+	cmd := NewCmdPatch(f, buf)
+	cmd.Flags().Set("namespace", "test")
+	cmd.Flags().Set("patch", `{"spec":{"type":"NodePort"}}`)
+	cmd.Flags().Set("output", "yaml")
```

Comment:
took a second look, fixed the print method, now it outputs yaml correctly even in test